src/mjolnir/ingest_transit.cc

```diff
@@ -0,0 +1,790 @@
+
+      for (const auto& stop : stops) {
+        if (stop.stop_id.empty()) {
+          continue;
+        }
+        gtfs::Id currStopId = stop.stop_id;
+        float x = stop.stop_lon;
+        float y = stop.stop_lat;
+
+        // add unique GraphId and stop
+        // might have to typecast these doubles into uint32_t
+        GraphId currId = GraphId(tile_level.tiles.TileId(y, x), 3, 0);
+
+        tileTransitInfo currTile;
+        if (tile_map.find(currId) != tile_map.end()) {
+          currTile = tile_map[currId];
+        } else {
+          currTile.graphId = currId;
+        }
+        currTile.tile_stops.insert({currStopId, feed_path});
+        gtfs::StopTimes currStopTimes = feed.get_stop_times_for_stop(currStopId);
+
+        for (const auto& stopTime : currStopTimes) {
+          // add trip, route, agency and service_id from stop_time
+          // could use std::set to not check if it already exists
+
+          // use a set instead to keep track of these points
+          if (stopTime.trip_id.empty() || !feed.get_trip(stopTime.trip_id)) {
+            continue;
+          }
+          gtfs::Trip currTrip = *(feed.get_trip(stopTime.trip_id));
+          if (currTrip.service_id.empty() || currTrip.route_id.empty() ||
+              !feed.get_route(currTrip.route_id)) {
+            continue;
+          }
+          currTile.tile_services.insert({currTrip.service_id, feed_path});
+          currTile.tile_trips.insert({currTrip.trip_id, feed_path});
+
+          currTile.tile_shapes.insert({currTrip.shape_id, feed_path});
+          // remember the shape id such that we can access it later
+          gtfs::Route currRoute = *(feed.get_route(currTrip.route_id));
+          currTile.tile_routes.insert({currRoute.route_id, feed_path});
+          currTile.tile_agencies.insert({currRoute.agency_id, feed_path});
+          // add to the queue of tiles
+        }
+        tile_map[currId] = currTile;
+      }
+    }
+  }
+  std::priority_queue<tileTransitInfo> prioritized;
+  for (auto it = tile_map.begin(); it != tile_map.end(); it++) {
+    prioritized.push(it->second);
+  }
+  return prioritized;
+}
+
+std::unordered_map<gtfs::Id, GraphId> write_stops(Transit& tile, const tileTransitInfo& tile_info) {
+
+  const auto& tile_stopIds = tile_info.tile_stops;
+  auto node_id = tile_info.graphId;
+  feedCache stopFeeds;
+
+  std::unordered_map<gtfs::Id, GraphId> stop_graphIds;
+  // loop through all stops inside the tile
+  for (const feedObject& feed_stop : tile_stopIds) {
+    const auto& tile_stopId = feed_stop.id;
+    const auto& feed = stopFeeds(feed_stop);
+    auto* node = tile.add_nodes();
+    gtfs::Stop tile_stop = *(feed.get_stop(tile_stopId));
+    node->set_lon(tile_stop.stop_lon);
+    node->set_lat(tile_stop.stop_lat);
+    node->set_type(static_cast<int>(tile_stop.location_type));
+    node->set_graphid(node_id);
+    node_id++;
+    // node->set_prev_graphid(); what is prev?
```

Comment:
used in convert_transit i believe

Reply:
also can you mark anything similar to this where we arent sure yet or havent implemented it with a `// TODO:` so its easier to search for in the code